src/lib/actions/RulesAction.ts

```diff
@@ -1,7 +1,39 @@
+import {CodeAnalyzer, CodeAnalyzerConfig, Rule, RuleSelection} from '@salesforce/code-analyzer-core';
 
+import {ConfigLoader} from '../loaders/ConfigLoader';
+import {EngineLoader} from '../loaders/EngineLoader';
+import {RuleViewer} from '../viewers/RuleViewer';
+
+export type RulesDependencies = {
+	configLoader: ConfigLoader;
+	engineLoader: EngineLoader;
+	viewer: RuleViewer;
+}
+
+export type RulesInput = {
+	'config-file'?: string;
+	'rule-selector': string[];
+}
 
 export class RulesAction {
-	public execute(_input: object): void {
+	private readonly dependencies: RulesDependencies;
+
+	public constructor(dependencies: RulesDependencies) {
+		this.dependencies = dependencies;
```

Comment:
This is just a stylistic thing.... but you don't really need a dependencies object. You could just pass in your dependencies directly. new RulesAction(configFactory, enginePluginFactory).